tools/src/wyvern/target/corewyvernIL/type/RefinementType.java

```diff
@@ -67,22 +72,52 @@ public RefinementType(List<ValueType> typeParams, ValueType base, HasLocation ha
                 // for debugging
                 base.checkWellFormed(ctx);
 
-                ToolError.reportError(ErrorMessage.CANNOT_APPLY_TYPE_PARAMETERS, this.getLocation(), base.toString());
+                ToolError.reportError(ErrorMessage.CANNOT_APPLY_GENERIC_ARGUMENTS, getLocation(), base.toString());
             }
-            int index = 0;
-            int declCount = st.getDeclTypes().size();
-            for (ValueType vt : typeParams) {
-                // advance the index to an AbstractTypeMember
-                while (index < declCount && !(st.getDeclTypes().get(index) instanceof AbstractTypeMember)) {
-                    index++;
+
+            final List<DeclType> stDeclTypes = st.getDeclTypes();
+            final Iterator<GenericArgument> genericArgumentIterator = genericArguments.iterator();
```

Comment:
`genericArguments` may be null, which isn't handled.

Reply:
Good catch, thanks!